Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -793,4 +784,41 @@ - (void) disconnect {
     _out = nil;
 }
 
++ (NSArray*) parseCookie: (NSString*) cookieStr {
+    Assert(cookieStr.length > 0, @"Trtying to parse empty cookie string");
+    
+    NSArray* rawAttrs = [cookieStr componentsSeparatedByString: @";"];
+    
+    NSMutableArray *attrs = [NSMutableArray arrayWithCapacity: [rawAttrs count]];
+    for (NSString* rawAttr in rawAttrs) {
+        // trims the attribute
+        NSString* attr = [rawAttr stringByTrimmingCharactersInSet: [NSCharacterSet whitespaceAndNewlineCharacterSet]];
+        
+        // replace comma with ^G(bell) in Date
+        if ([attr hasPrefix: @"Expires"] || [attr hasPrefix: @"expires"]) {
+            NSRange range = [attr rangeOfString: @","];
+            if (range.location != NSNotFound)
+                attr = [attr stringByReplacingCharactersInRange: range withString: @"^G"];
+        }
+        [attrs addObject: attr];
+    }
+    
+    // separate out with cookie boundaries
+    NSArray* rawCookies = [[attrs componentsJoinedByString: @";"] componentsSeparatedByString: @","];
+    NSMutableArray *cookies = [NSMutableArray arrayWithCapacity: [rawCookies count]];
+    for (NSString* rawCookie in rawCookies) {
+        // trim the cookie
+        NSString* cookie = [rawCookie stringByTrimmingCharactersInSet: [NSCharacterSet whitespaceAndNewlineCharacterSet]];
```

Comment:
I think you don't need to trim it again, right?

Reply:
`a1=b1,    a2=b2` this space between the cookies are trimmed here.